addons/mail/static/src/new/public/discuss_public.js

```diff
@@ -24,13 +24,18 @@ export class DiscussPublic extends Component {
         useEffect(
             (welcome) => {
                 if (!welcome) {
-                    // Change the URL to avoid leaking the invitation link.
-                    window.history.replaceState(
-                        window.history.state,
-                        null,
-                        `/discuss/channel/${this.thread.id}${window.location.search}`
+                    if (this.props.data.discussPublicViewData.isChannelTokenSecret) {
+                        // Change the URL to avoid leaking the invitation link.
+                        window.history.replaceState(
+                            window.history.state,
+                            null,
+                            `/discuss/channel/${this.thread.id}${window.location.search}`
+                        );
+                    }
+                    this.threadService.setDiscussThread(
+                        this.thread,
+                        !this.props.data.discussPublicViewData.isChannelTokenSecret
```

Comment:
Isn't `active_id` never useful on the public page? I think this could stay `false`.